common/src/KokkosFFT_utils.hpp

```diff
@@ -85,38 +86,82 @@ auto convert_negative_shift(const ViewType& view, int _shift, int _axis) {
   return std::tuple<int, int, int>({shift0, shift1, shift2});
 }
 
-template <typename T>
-bool is_found(std::vector<T>& values, const T& key) {
+template <typename ContainerType, typename ValueType>
+bool is_found(ContainerType& values, const ValueType& key) {
+  using value_type = KokkosFFT::Impl::base_container_value_type<ContainerType>;
+  static_assert(std::is_same_v<value_type, ValueType>,
+                "Container value type must match ValueType");
   return std::find(values.begin(), values.end(), key) != values.end();
 }
 
-template <typename T>
-bool has_duplicate_values(const std::vector<T>& values) {
-  std::set<T> set_values(values.begin(), values.end());
+template <typename ContainerType>
+bool has_duplicate_values(const ContainerType& values) {
+  using value_type = KokkosFFT::Impl::base_container_value_type<ContainerType>;
+  std::set<value_type> set_values(values.begin(), values.end());
   return set_values.size() < values.size();
 }
 
-template <typename IntType, std::enable_if_t<std::is_integral_v<IntType>,
-                                             std::nullptr_t> = nullptr>
-bool is_out_of_range_value_included(const std::vector<IntType>& values,
-                                    IntType max) {
+template <
+    typename ContainerType, typename IntType,
+    std::enable_if_t<std::is_integral_v<IntType>, std::nullptr_t> = nullptr>
+bool is_out_of_range_value_included(const ContainerType& values, IntType max) {
```

Comment:
Not related but for this function you can do an early exit in the for loop.

Reply:
That is true. Fixed accordingly